src/angular-multi-select-utils.js

```diff
@@ -49,7 +49,9 @@ angular_multi_select_utils.factory('angularMultiSelectUtils', [
 			 * array or undefined.
 			 */
 			if (typeof(str) === 'string') {
-				return str.split(",").map(s => s.replace(/^\s+|\s+$/g, ''));
+				return str.split(",")
+					.map(s => s.replace(/^\s+|\s+$/g, ''))
+					.filter(s => s.length > 0);
```

Comment:
I'm not really sure this will work as you expect (?). Let's say that the `str` is `id, 1, id, 2, id, , id, 4` (note the missing `3`).

My version will output `["id", "1", "id", "2", "id", "", "id", "4"]`, which is a fine-balanced array for `parse_pairs` function. As soon as `parse_pairs` reaches the third pair of values, it will get `[id, ]`, pass it to Loki and forget about it (valid behavior).

Your version will output `["id", "1", "id", "2", "id", "id", "4"]`, which is an unbalanced array. The third pair of values is `[id, id]`, and the forth pair of values is incomplete (`[4]`).

If you really want to skip empty values (either key or value), you should remove both elements of the pair. In this specific example, your code should return `["id", "1", "id", "2", "id", "4"]`.